### ISSUE TITLE
QUESTION ReBar setting in BIOS

### ISSUE THREAD

**Issue by xtomasnemec** [xCuri0/ReBarUEFI#203]
Is there a way of adding the ReBar setting to the BIOS (AMI) trough AMIBCP?

**Comment by xCuri0** [comment]
Not to my knowledge but it should be technically possible.

You will first need to modify ReBarDxe to read an offset in the Setup variable which will be the easiest part.

The hard part will be modifying the UEFI Internal Form Representation (IFR) to add an entry for Resizable BAR that controls the offset at in the Setup variable that you previously set. But no such IFR editor currently exists

@NikolajSchlej I've seen your work on IFR Extractor and UEFITool do you think it would be possible to modify the UEFI IFR structures to add (or replace if size limitation) a new entry ?

**Comment by NikolajSchlej** [comment]
Not via AMIBCP, no, but a DXE driver can indeed publish its own HII structures without a need to modify anything. Intel RAID OROMs do so, so do network card drivers and some other stuff. You can add this to ReBarDxe by reading this part of Driver Writers Guide: https://github.com/tianocore-docs/edk2-UefiDriverWritersGuide/tree/master/12_uefi_driver_configuration

**Comment by NikolajSchlej** [comment]
This assumes that ReBarDxe can be loaded before BIOS Setup is shown, which can be achieved by either adding it to DXE volume, or using DriverXXXX mechanism to load it in early BDS.

**Comment by xCuri0** [comment]
@NikolajSchlej Didn't know this existed thanks for sharing.

And yes it's loaded very early during boot before PCI resources are even allocated

**Comment by xtomasnemec** [comment]
Thanks for the info

**Comment by xtomasnemec** [comment]
I took another look at it. Couldn't I just add an option to the Setup section with the offset of the RebarState variable?
I'd just need to figure out the offset somehow.

**Comment by xCuri0** [comment]
@xtomasnemec no, just do it the way mentioned by [NikolajSchlej](https://github.com/NikolajSchlej)
